packages/flutter/lib/src/rendering/object.dart

```diff
@@ -472,6 +472,11 @@ class PaintingContext {
       childContext._stopRecordingIfNeeded();
     } else {
       if (elevation != 0) {
+        // The drawShadow call doesn't add the region of the shadow to the
+        // picture's bounds, so we draw a hardcoded amount of extra space to
+        // account for the maximum potential area of the shadow.
+        canvas.drawRect(offsetBounds.inflate(20.0),
+                        new Paint()..color=const Color(0));
```

Comment:
Cool, in that case the TODO can just be "Remove this once skia does it for us". :-)